app/controllers/time_tracker_controller.rb

```diff
@@ -1,50 +1,48 @@
 # frozen_string_literal: true
 
-# rubocop:disable Style/IfUnlessModifier, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/AbcSize, Metrics/MethodLength
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/MethodLength
 class TimeTrackerController < TrackyController
-  def start
-    # Early return if timer already running
-    if @current_timer_session
-      render :update, layout: false and return
-    end
+  def upsert
+    @current_timer_session = TimerSession.active.find_by(user: User.current)
 
-    @current_timer_session = SessionCreator.new(@current_user, timer_params, params[:commit]).create
+    return start_timer unless @current_timer_session
+    return cancel_timer if params[:cancel]
```

Comment:
```suggestion
    return cancel_timer if params[:cancel].present?
```

Reply:
Keep cancel as separate endpoint? Look into TimerSession attributes for cancelling.

Follow Rails conventions for params.